src/main/kotlin/com/linchpino/core/service/EmailService.kt

```diff
@@ -0,0 +1,82 @@
+package com.linchpino.core.service
+
+import com.linchpino.core.entity.Interview
+import com.linchpino.core.enums.AccountStatusEnum
+import jakarta.mail.internet.InternetAddress
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.core.env.Environment
+import org.springframework.mail.javamail.JavaMailSender
+import org.springframework.mail.javamail.MimeMessageHelper
+import org.springframework.stereotype.Service
+import org.thymeleaf.context.Context
+import org.thymeleaf.spring6.SpringTemplateEngine
+
+@Service
+class EmailService(
+    private val emailSender: JavaMailSender,
+    private val templateEngine: SpringTemplateEngine,
+    private val environment: Environment
+) {
+    @Value("\${application.url}")
+    var applicationUrl: String? = null
+
+    fun sendingInterviewInvitationEmailToJobSeeker(interview: Interview) {
+        val fullName = "${interview.jobSeekerAccount!!.firstName} ${interview.jobSeekerAccount!!.lastName}"
+        val finalName = fullName.takeIf { it.isEmpty() } ?: "Jobseeker"
+        val date = interview.timeSlot!!.fromTime.toLocalDate()
+        val time = interview.timeSlot!!.fromTime.toLocalTime()
+        val zone = interview.timeSlot!!.fromTime.getZone()
+        val isJobSeekerIsNotActivated = interview.jobSeekerAccount!!.status != AccountStatusEnum.ACTIVATED
+        val jobSeekerExternalId = interview.jobSeekerAccount!!.externalId.toString()
+
+        val templateContextData = mapOf(
+            "fullName" to finalName,
+            "date" to date,
+            "time" to time,
+            "timezone" to zone,
+            "isJobSeekerIsNotActivated" to isJobSeekerIsNotActivated,
+            "applicationUrl" to applicationUrl.toString(),
+            "interviewId" to interview.id.toString(),
+            "jobSeekerExternalId" to jobSeekerExternalId
+        )
+
+        val mailFrom: String? = environment.getProperty("spring.mail.properties.mail.smtp.from")
+        val mailFromName: String = environment.getProperty("mail.from.name", "Linchpino")
+
```

Comment:
Use @Value annotation instead of calling getProperty on environment.

Reply:
Done